lib/internal/modules/run_main.js

```diff
@@ -166,9 +165,7 @@ function executeUserEntryPoint(main = process.argv[1]) {
     wrapModuleLoad(main, null, true);
   } else {
     const mainPath = resolvedMain || main;
-    if (mainURL === undefined) {
-      mainURL = pathToFileURL(mainPath).href;
-    }
+    const mainURL = getOptionValue('--entry-url') ? new URL(mainPath, getCWDURL()).href : pathToFileURL(mainPath).href;
```

Comment:
We can avoid repeating the `.href` here, and we should probably rename `mainPath`.
```suggestion
    const mainURL = getOptionValue('--entry-url') ? new URL(mainPathOrURL, getCWDURL()) : pathToFileURL(mainPathOrURL);
```